web-marketplace/src/hooks/projects/useProjectsWithOrders.ts

```diff
@@ -149,18 +149,30 @@ export function useProjectsWithOrders({
     projects = projectsData?.projects;
   }
 
+  const offChainIds = allOffChainProjects.map(project => project.id);
+  const clientProjects = projects?.filter(project =>
+    IS_TERRASOS ? offChainIds.includes(project.id) : true,
```

Comment:
we are comparing off chain uuids to on chain project ids here, why?

Reply:
That was a mistake. I got confused because we didn't have the onChain ID on the object. I had to change the query and the code a bit, but it should be good now. Projects with the Terrasos type are displayed correctly on the preview.